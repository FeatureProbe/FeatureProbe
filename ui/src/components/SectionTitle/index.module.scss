.title {
  display: flex;
  align-items: center;
  height: 28px;
}

.divider {
  width: 4px;
  height: 15px;
  background-color: #556ee6;
}

.name {
  margin-left: 12px;
  font-size: 14px;
  color: #212529;
  font-family: HelveticaNeue-Medium;
}

.icon-question {
  color: #74788d;
  vertical-align: middle;
  margin-left: 4px;
}