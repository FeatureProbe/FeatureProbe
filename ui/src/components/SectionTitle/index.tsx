import { Popup } from 'semantic-ui-react';
import Icon from 'components/Icon';
import styles from './index.module.scss';

interface IProp {
  title: string;
  tooltipText?: string;
  showTooltip?: boolean;
}

const SectionTitle = (props: IProp) => {
  const { title, tooltipText, showTooltip } = props;
	return (
    <div className={styles.title}>
      <div className={styles.divider}></div>
      <div className={styles.name}>{ title }</div>
      {
        showTooltip && (
          <Popup
            inverted
            style={{ opacity: 0.8 }}
            trigger={<Icon type='question' customclass={styles['icon-question']} />}
            content={ tooltipText }
            position='top center'
          />
        )
      }
    </div>
	);
};

export default SectionTitle;