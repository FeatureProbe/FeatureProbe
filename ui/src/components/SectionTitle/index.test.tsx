import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import SectionTitle from '.';

it('PutAwayMenu snapshot', (done) => {
  (async () => {
    const { asFragment } = render(<SectionTitle title="title" tooltipText="tooltipText" showTooltip={true} />);
    expect(asFragment()).toMatchSnapshot();
    done();
  })();
});
