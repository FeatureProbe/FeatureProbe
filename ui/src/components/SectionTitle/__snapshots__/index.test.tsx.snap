// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PutAwayMenu snapshot 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <div
      class="divider"
    />
    <div
      class="name"
    >
      title
    </div>
    <span
      class="iconfont icon-question icon-question"
      customclass="icon-question"
      type="question"
    />
  </div>
</DocumentFragment>
`;
